src/ipc/utils/eigen_ext.tpp

```diff
@@ -84,12 +87,27 @@ project_to_psd(
     }
     Eigen::DiagonalMatrix<double, Eigen::Dynamic> D(eigensolver.eigenvalues());
     // Save a little time and only project the negative values
-    for (int i = 0; i < A.rows(); i++) {
-        if (D.diagonal()[i] < 0.0) {
-            D.diagonal()[i] = 0.0;
-        } else {
-            break;
+    switch (type)
```

Comment:
Can you place the switch inside the for loop to reduce code redundancy? Branch prediction should mitigate the performance difference.